lib/hydra/works/services/determine_mime_type.rb

```diff
@@ -0,0 +1,24 @@
+module Hydra::Works
+  class DetermineMimeType
+    # Determines the mime type for a given file
+    # @param [IO, File, Rack::Multipart::UploadedFile, #read] file
+    # @param [String, NilClass] original_name of the file
+    # @return [String]
+    class << self
```

Comment:
@mjgiarlo I was going by what's currently in the repo. The majority use `self.call` without an additional initializer, but I share your opinions. The primary motivation behind doing this was to avoid having to use Rspec's `allow_any_instance_of` because the original `determine_mime_type` method was nested deep in a private method, in an internal class. I  found it too difficult to mock that properly, and it highlighted the fact that `AddFileToFileSet` was too concerned with mime types and names.

True, it is basically a module, but in a service interface it's more accessible to another application. Of course this is a bit YAGNI, but like I said, the primary motivation for this was to get around mocking issues and adhering more to single-responsibility principles. I could have gone with the module design, but a service, I think, is an extra bonus in this case.